File: CodeForces/Contest/34/D.cpp
#include <algorithm>
#include <cstdio>
#include <map>

using namespace std;

long long n, x, a, b, c, ans, ttl;
map <int, int> freq;

long long bnyk(long long x) {
	if(freq.count(x)) {
		return freq[x];
	} else {
		return 0;
	}
}

int main() {
	scanf("%lld", &n);
	for(long long i = 1; i <= n; i++) {
		scanf("%lld", &x);
		a = bnyk(x - 1);
		b = bnyk(x);
		c = bnyk(x + 1);
		ans += x * (i - 1 - a - b - c) - (ttl - (x - 1) * a - x * b - (x + 1) * c);
		ttl += x;
		freq[x]++;
		if(i % 1000 == 0) {
			printf("%d\n", i);
		}
	}
	printf("%lld\n", ans);
	return 0;
}

File: CodeForces/Contest/34/D.py
def bnyk(x):
	if(x in freq.keys()):
		return freq[x]
	else:
		return 0;

freq = {

}


n = int(input())
list = input().split()
ans = 0
ttl = 0
for i in range(n):
	x = int(list[i])
	a = bnyk(x - 1)
	b = bnyk(x)
	c = bnyk(x + 1)
	ans += x * (i - a - b - c) - (ttl - (x - 1) * a - x * b - (x + 1) * c)
	ttl += x
	if(x in freq.keys()):
		freq[x] += 1
	else:
		freq[x] = 1
print("%d\n" % ans)